backend/app/services/model_service.py

```diff
@@ -92,19 +90,90 @@ def get_loaded_inference_model(self, force_reload: bool = False) -> LoadedModel
-        return model
+    def get_model_by_id(self, project_id: UUID, model_id: UUID) -> ModelSchema:
+        """
+        Get a model by its ID within a specific project.
+
+        Retrieves a model revision from the specified project by matching the model ID.
+        The method first validates that the project exists, then searches through the
+        project's model revisions to find the one with the matching ID.
+
+        Args:
+            project_id (UUID): The unique identifier of the project containing the model.
+            model_id (UUID): The unique identifier of the model to retrieve.
+
+        Returns:
+            ModelSchema: The model schema object containing the model's information.
+
+        Raises:
+            ResourceNotFoundError: If the project with the given project_id does not exist,
+                or if no model with the given model_id is found within the project.
+        """
+        with get_db_session() as db:
+            project_repo = ProjectRepository(db)
+            project = project_repo.get_by_id(str(project_id))
+            if not project:
+                raise ResourceNotFoundError(ResourceType.PROJECT, str(project_id))
+            model = next((self._mapper.to_schema(m) for m in project.model_revisions if m.id == str(model_id)), None)
+            if not model:
+                raise ResourceNotFoundError(ResourceType.MODEL, str(model_id))
+            return model
 
     @parent_process_only
-    def delete_model_by_id(self, model_id: UUID) -> None:
-        """Delete a model by its ID"""
+    def delete_model_by_id(self, project_id: UUID, model_id: UUID) -> None:
+        """
+        Delete a model by its ID from a specific project.
+
+        Permanently removes a model revision from the specified project. The method
+        first validates that the project exists, then attempts to delete the model
+        from the database. This operation is restricted to the parent process only.
+
+        Args:
+            project_id (UUID): The unique identifier of the project containing the model.
+            model_id (UUID): The unique identifier of the model to delete.
+
+        Returns:
+            None
+
+        Raises:
+            ResourceNotFoundError: If the project with the given project_id does not exist,
+                or if no model with the given model_id is found.
+            ResourceInUseError: If the model cannot be deleted due to integrity constraints
+                (e.g., the model is referenced by other entities).
+        """
         with get_db_session() as db:
-            self._persistence.delete_by_id(model_id, db)
+            project_repo = ProjectRepository(db)
+            if not project_repo.exists(str(project_id)):
+                raise ResourceNotFoundError(ResourceType.PROJECT, str(project_id))
+            model_rev_repo = ModelRevisionRepository(db)
+            try:
+                deleted = model_rev_repo.delete(str(model_id))
```

Comment:
You should also delete the model binaries

Reply:
Since BYOM is no longer supported and model artifacts are now placed manually for testing, I suggest we address this once model tuning is finalized. 
Adding a `TODO` comment for now.